File: src/main/java/com/blackcowmoo/moomark/auth/repository/UserRepository.java
package com.blackcowmoo.moomark.auth.repository;

import java.util.Optional;

import com.blackcowmoo.moomark.auth.model.AuthProvider;
import com.blackcowmoo.moomark.auth.model.entity.User;

import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, Long> {
  Optional<User> findByEmail(String email);

  Optional<User> findByName(String name);

  Optional<User> findByEmailAndAuthProvider(String email, AuthProvider authProvider);

  Optional<User> findByEmailAndName(String email, String name);

  Optional<User> findByNicknameLike(String nickname);

}

File: src/main/kotlin/com/blackcowmoo/moomark/auth/repository/UserRepository.kt
package com.blackcowmoo.moomark.auth.repository

import com.blackcowmoo.moomark.auth.model.AuthProvider
import com.blackcowmoo.moomark.auth.model.entity.User
import com.blackcowmoo.moomark.auth.model.entity.UserId
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, UserId> {
  fun findByIdAndAuthProvider(id: String, authProvider: AuthProvider): User?
}
